src/query/management/src/role/role_api.rs

```diff
@@ -44,11 +46,37 @@ pub trait RoleApi: Sync + Send {
     /// from: the role that currently owns the object, it could be empty on the first time
     /// to: the role that will own the object, to.0 is the owner role name, to.1 is the role details
     /// None RoleInfo means the role is built-in role, could only update grant object metadata
+    ///
+    /// Grant ownership used when create new object, contains two step:
+    /// 1. grant ownership privilege obj to role,
+    /// 2. kv api upsert new owner object key.
+    ///
+    /// Note: if role is `account_admin` no need to grant
     #[allow(clippy::ptr_arg)]
     async fn grant_ownership(&self, object: &OwnershipObject, role: &str) -> Result<()>;
 
     /// Remember to call this method when you dropped a OwnerObject like table/database/stage/udf.
-    async fn revoke_ownership(&self, object: &OwnershipObject) -> Result<()>;
+    /// Revoke ownership used when drop old object, contains two step:
+    /// 1. revoke ownership privilege obj to new role,
+    /// 2. kv api delete old owner object key.
+    ///
+    /// Note: if role is `account_admin` or None no need to revoke
+    async fn revoke_ownership(&self, object: &OwnershipObject, role: Option<String>) -> Result<()>;
```

Comment:
`role` -> `old_role` ?

as it intends to drop it in the role's grant set in a txn.

Reply:
Now will get ownership inside revoke ownership